File: src/main/java/ordilov/lissn/config/AppProperties.java
package ordilov.lissn.config;

import java.util.List;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.ConstructorBinding;

@Getter
@ConstructorBinding
@RequiredArgsConstructor
@ConfigurationProperties(prefix = "auth")
public class AppProperties {

  private final String tokenSecret;
  private final Long tokenExpiration;
  private final Long tokenRefreshExpiration;
  private final List<String> authorizedRedirectUrls;

}

File: src/main/java/ordilov/lissn/config/AppProperties.kt
package ordilov.lissn.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConstructorBinding
@ConfigurationProperties(prefix = "auth")
data class AppProperties(
    var tokenSecret: String,
    var tokenExpiration: Long,
    var tokenRefreshExpiration: Long,
//    val authorizedRedirectUrls: List<String>
) {
    constructor() : this("", 0, 0)

    init {
        println("tokenSecret $tokenSecret")
    }
}